controllers/openstackdataplanerole_controller.go

```diff
@@ -68,9 +78,190 @@ func (r *OpenStackDataPlaneRoleReconciler) Reconcile(ctx context.Context, req ct
+	}
+	if instance.Spec.NodeTemplate.Managed {
+		overcloud.Vars["managed"] = instance.Spec.NodeTemplate.Managed
+	}
+	if instance.Spec.NodeTemplate.ManagementNetwork != "" {
+		overcloud.Vars["management_network"] = instance.Spec.NodeTemplate.ManagementNetwork
+	}
+	if instance.Spec.NodeTemplate.NetworkConfig.Template != "" {
+		overcloud.Vars["network_config"] = fmt.Sprintf("{template: %s}", instance.Spec.NodeTemplate.NetworkConfig.Template)
+	}
+	if len(instance.Spec.NodeTemplate.Networks) > 0 {
+		var network string // the resulting string containing each network
+		for _, netMap := range instance.Spec.NodeTemplate.Networks {
+			if netMap.FixedIP != "" && netMap.Network == "" {
+				network += fmt.Sprintf("{%s: %s},", "fixedIP", netMap.FixedIP)
+			}
+			if netMap.FixedIP != "" && netMap.Network != "" {
+				network += fmt.Sprintf("{%s: %s, %s: %s},",
+					"fixedIP", netMap.FixedIP, "network", netMap.Network)
+			}
+		}
+		overcloud.Vars["networks"] = fmt.Sprintf("[%s]", strings.TrimSuffix(network, ","))
+	}
+
+	ansibleVarsData := make(map[string]interface{})
+	err = yaml.Unmarshal([]byte(instance.Spec.NodeTemplate.AnsibleVars), ansibleVarsData)
+	if err != nil {
+		return "", err
+	}
+	for key, value := range ansibleVarsData {
+		overcloud.Vars[key] = value
+	}
+
+	compute := overcloud.AddChild(ansible.MakeGroup("Compute"))
+	for _, node := range nodes {
+		host := compute.AddHost(node.Name)
+		if node.Spec.Node.AnsibleUser != "" {
+			host.Vars["ansible_user"] = node.Spec.Node.AnsibleUser
+		}
+		if node.Spec.Node.AnsiblePort > 0 {
+			host.Vars["ansible_port"] = node.Spec.Node.AnsiblePort
+		}
+		if node.Spec.Node.Managed {
+			host.Vars["managed"] = node.Spec.Node.Managed
+		}
+		if node.Spec.Node.ManagementNetwork != "" {
+			host.Vars["management_network"] = node.Spec.Node.ManagementNetwork
+		}
+		if node.Spec.Node.NetworkConfig.Template != "" {
+			host.Vars["network_config"] = fmt.Sprintf("{template: %s}", node.Spec.Node.NetworkConfig.Template)
+		}
+		if len(node.Spec.Node.Networks) > 0 {
+			var network string // the resulting string containing each network
+			for _, netMap := range node.Spec.Node.Networks {
+				if netMap.FixedIP != "" && netMap.Network == "" {
+					network += fmt.Sprintf("{%s: %s},", "fixedIP", netMap.FixedIP)
+				}
+				if netMap.FixedIP != "" && netMap.Network != "" {
+					network += fmt.Sprintf("{%s: %s, %s: %s},",
+						"fixedIP", netMap.FixedIP, "network", netMap.Network)
+				}
+			}
+			host.Vars["networks"] = fmt.Sprintf("[%s]", strings.TrimSuffix(network, ","))
+		}
+		if node.Spec.AnsibleHost == "" {
+			hostName = node.Spec.HostName
+		} else {
+			hostName = node.Spec.AnsibleHost
+		}
+		host.Vars["ansible_host"] = hostName
+
+		ansibleVarsData := make(map[string]interface{})
+		err = yaml.Unmarshal([]byte(node.Spec.Node.AnsibleVars), ansibleVarsData)
+		if err != nil {
+			return "", err
+		}
+		for key, value := range ansibleVarsData {
+			host.Vars[key] = value
+		}
+	}
```

Comment:
are these the bits that you'd refactor into a common function? something like:

`ansibleVars := resolveAnsibleVars(node)`

?

Reply:
yes, I'll try to have common functions for role and node